### PR TITLE
Type cast falsy boolean symbols on boolean attribute as false

### LINKED ISSUE #35676: PostgreSQL adapter: breaking change in boolean parameter parsing in 5.2.1
Between 5.2.0 and 5.2.1, boolean parameter parsing in queries saw a breaking change. Consider the following:

```ruby
Post.where(published: :false)
```

In 5.2.0, this creates the following query:

```sql
SELECT "posts".* FROM "posts" WHERE "posts"."published" = 'false'
```

In 5.2.1, the query is instead:

```sql
SELECT "posts".* FROM "posts" WHERE "posts"."published" = TRUE
```

the result of which is the opposite of the former. (Postgres interprets `"false"` as `FALSE`.)

While I think that having any truthy value parse to `TRUE` is a sensible default, it is a breaking change that could have potentially disastrous consequences.

### Steps to reproduce

```ruby
ActiveRecord::Schema.define do
  create_table :posts, force: true do |t|
    t.boolean :published
  end
end

class Post < ActiveRecord::Base
end

class BugTest < Minitest::Test
  def test_boolean_querying
    Post.create!(published: true)

    assert_equal Post.where(published: false).count, Post.where(published: :false).count
  end
end
```

### Expected behavior

Query results are consistent between patch (and minor) versions.

### Actual behavior

Query results change between patch (and minor) versions.

### System configuration
**Rails version**:

5.2.1

**Ruby version**:

2.6.2

### PR DESCRIPTION
**Pull request by kamipo**
Before 34cc301, type casting by boolean attribute when querying is a
no-op, so finding by truthy boolean string (i.e.
`where(value: "true") # => value = 'true'`) didn't work as expected
(matches it to FALSE in MySQL #32624). By type casting is ensured, a
value on boolean attribute is always serialized to TRUE or FALSE.

In PostgreSQL, `where(value: :false) # => value = 'false'` was a valid
SQL, so 34cc301 is a regresson for PostgreSQL since all symbol values
are serialized as TRUE.

I'd say using `:false` is mostly a developer's mistake (user's input
basically comes as a string), but `:false` on boolean attribute is
serialized as TRUE is not a desirable behavior for anybody.

This allows falsy boolean symbols as false, i.e.
`klass.create(value: :false).value? # => false` and
`where(value: :false) # => value = FALSE`.

Fixes #35676.

### CODE REVIEW

**Review by rafaelfranca — APPROVED** [review]

**Comment by kamipo** [comment]
Should this be backported to 5-2-stable?
This fixes #35676, but also changes the behavior of casting `klass.create(value: :false).value?` from true to false.

**Comment by rafaelfranca** [comment]
Was it `false` before 5.2.3 or `true`?

**Comment by kamipo** [comment]
`klass.create(value: :false).value?` is `true` since 5.0 (`nil` before 5.0).

**Comment by rafaelfranca** [comment]
hmmm. Let's backport and see if people complain about it during the rc

**Comment by kamipo** [comment]
Sure, I've backported in 724e1f4efd0a33e83aa9ee5b6067e21e88d739ec.
